packages/flutter/lib/src/widgets/pop_scope.dart

```diff
@@ -8,12 +8,52 @@ import 'framework.dart';
 import 'navigator.dart';
 import 'routes.dart';
 
+/// Deprecated, use [PopInvokedWithResultCallback] instead
+@Deprecated(
+  'Use PopInvokedWithResultCallback instead. '
+  'This feature was deprecated after v3.22.0-12.0.pre.',
+)
+typedef PopInvokedCallback = void Function(bool didPop);
+
+/// Deprecated, use [PopScopeWithResult] instead.
+@Deprecated(
+  'Use PopScopeWithResult instead. '
+  'This feature was deprecated after v3.22.0-12.0.pre.',
+)
+class PopScope extends PopScopeWithResult<Object?> {
+  /// Deprecated, use [PopScopeWithResult] instead.
+  @Deprecated(
+    'Use PopScopeWithResult instead. '
+    'This feature was deprecated after v3.22.0-12.0.pre.',
+  )
+  const PopScope({
+    super.key,
+    super.canPop = true,
+    PopInvokedCallback? onPopInvoked,
+    required super.child,
+  }) : _onPopInvoked = onPopInvoked;
+
+  @override
+  PopInvokedWithResultCallback<Object?>? get onPopInvoked {
+    if (_onPopInvoked == null) {
+      return null;
+    }
+    return (bool didPop, Object? result) => _onPopInvoked(didPop);
+  }
+  final PopInvokedCallback? _onPopInvoked;
+}
+
 /// Manages back navigation gestures.
 ///
+/// The generic type should match or be supertype of the generic type of the
```

Comment:
Missing "a" and a slight rewording:

```suggestion
/// The generic type should match or be a supertype of the generic type of the
/// enclosing [Route]. For example, if the enclosing Route is a `MaterialPageRoute<int>`,
/// you can define [PopScopeWithResult] with `int` or any supertype of `int`.
```